tests/units/Impact.class.php

```diff
@@ -0,0 +1,287 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2019 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+*/
+
+namespace tests\units;
+
+class Impact extends \GLPITestCase {
+
+   private function clearImpactItems() {
+      global $DB;
+
+      $DB->delete(\ImpactItem::getTable(), [true]);
+      $DB->delete(\ImpactRelation::getTable(), [true]);
+      $DB->delete(\ImpactCompound::getTable(), [true]);
+   }
+
+   public function testGetTabNameForItem_notCommonDBTM() {
+      $this->clearImpactItems();
+      $impact = new \Impact();
+
+      $this->exception(function() use ($impact) {
+         $notCommonDBTM = new \Impact();
+         $impact->getTabNameForItem($notCommonDBTM);
+      })->isInstanceOf(\InvalidArgumentException::class);
+   }
+
+   public function testGetTabNameForItem_notEnabledOrITIL() {
+      $this->clearImpactItems();
+      $impact = new \Impact();
+
+      $this->exception(function() use ($impact) {
+         $notEnabledOrITIL = new \ImpactCompound();
+         $impact->getTabNameForItem($notEnabledOrITIL);
+      })->isInstanceOf(\InvalidArgumentException::class);
+   }
+
+   public function testGetTabNameForItem_tabCountDisabled() {
+      $this->clearImpactItems();
+      $oldSession = $_SESSION['glpishow_count_on_tabs'];
+      $_SESSION['glpishow_count_on_tabs'] = false;
+
+      $impact = new \Impact();
+      $computer = new \Computer();
+
+      $this->string($impact->getTabNameForItem($computer))
```

Comment:
To be sure to put back `$_SESSION` to original state if test is failing, you should restore its value before doing asserts.
```
$tabname = $impact->getTabNameForItem($computer);
$_SESSION['glpishow_count_on_tabs'] = $oldSession;
$this->string($tabname)->isEqualTo("Impacts ");
```

This remark is valid for all tests.